src/WhereConditions/WhereHasConditionsDirective.php

```diff
@@ -52,17 +52,15 @@ public function handleBuilder($builder, $whereConditions)
             return $builder;
         }
 
-        return $builder->whereHas(
-            $this->getRelationName(),
-            function ($builder) use ($whereConditions): void {
-                // This extra nesting is required for the `OR` condition to work correctly.
-                $builder->whereNested(
-                    function ($builder) use ($whereConditions): void {
-                        $this->handleWhereConditions($builder, $whereConditions);
-                    }
-                );
-            }
-        );
+        return $this->handleWhereConditions($builder, [
+            'HAS' => [
+                'relation'  => $this->getRelationName(),
+                // TODO: There should be a way to convert operator to its value from the code; WhereConditionsServiceProvider::DEFAULT_HAS_OPERATOR
+                'operator'  => '>=',
```

Comment:
This value is an implementation detail of the `Operator` interface.

Reply:
Sorry if it's another silly question. But how exactly I use it?

Reply:
Users are free to implement their own version of the `Operator` interface, with `SQLOperator` being just one possible implementation. The customs implementation can be bound through a service provider.

The implementation of the where directives must not assume that `SQLOperator` is used, that may not be the case. Thus, we must make no assumptions about the internal values of the operators.

Reply:
> Note that this is a breaking change and will have to target the upcoming `v5` release.

Found a better alternative without it becoming a breaking change,